Rubberduck.RibbonDispatcher/Sample Ribbons/SampleRibbon/src/RibbonModel.cls

```diff
@@ -0,0 +1,131 @@
+    Set mCustomOptions = Factory.NewRibbonDropDown("ButtonOptions2", True, False)
+    mCustomOptions.AddItem Factory.NewSelectableItem("LabelOnly")
+    mCustomOptions.AddItem Factory.NewSelectableItem("ImageOnly")
+    mCustomOptions.AddItem Factory.NewSelectableItem("LabelAndImage")
+    
+    Set this.StandardGroup = Factory.NewRibbonGroup("StandardButtonsGroup")
+    Set mStandardButton1 = Factory.NewRibbonButtonMso("StandardButton1", True, True, rdRegular, "RefreshAll")
+    Set mStandardButton2 = Factory.NewRibbonButtonMso("StandardButton2", True, True, rdRegular, "Refresh")
+    Set mStandardOptions = Factory.NewRibbonDropDown("ButtonOptions", True, True)
+    Set mShowAdvancedToggle = Factory.NewRibbonToggle("ShowAdvancedToggle", True, True)
+    
+    mShowAdvancedToggle.OnToggled False
+    mStandardOptions.OnActionDropDown "", 2
+
+    mCustomOptions.SelectedItemId = "LabelAndImage"
+    mCustomOptions.IsEnabled = mCustomSizeToggle.IsPressed
+    
+    Set Initialize = Me
+XT: Exit Function
+EH: ReraiseError Err, ModuleName & "Initialize"
+    Resume XT
+    Resume Next
+End Function
+
+Private Sub mBrandingButton_Clicked()
+    MsgBox "Quack!", vbOKOnly Or vbInformation, "Rubberduck Ribbon Refresh"
+End Sub
+
+Private Sub mCustomButton1_Clicked()
+    DefaultButtonAction mCustomButton1
+End Sub
+Private Sub mCustomButton2_Clicked()
+    DefaultButtonAction mCustomButton2
+End Sub
+Private Sub mCustomButton3_Clicked()
+    DefaultButtonAction mCustomButton3
+End Sub
+Private Sub mCustomOptions_SelectionMade(ByVal ItemId As String, ByVal ItemIndex As Long)
+    SetButtonView ItemIndex, mCustomButton1, mCustomButton2, mCustomButton3
+End Sub
+
+Private Sub mCustomSizeToggle_Toggled(ByVal IsPressed As Boolean)
+    ToggleCustomSize Not IsPressed, mCustomButton1, mCustomButton2, mCustomButton3
+End Sub
+Private Sub ToggleCustomSize(ByVal IsLarge As Boolean, ParamArray Buttons())
+    Dim i As Long
+    For i = LBound(Buttons) To UBound(Buttons)
+        Buttons(i).Size = IIf(IsLarge, rdLarge, rdRegular)
+    Next i
+    mCustomOptions.IsEnabled = Not IsLarge
+End Sub
+
+Private Sub mStandardButton1_Clicked()
+    DefaultButtonAction mStandardButton1
+End Sub
+Private Sub mStandardButton2_Clicked()
+    DefaultButtonAction mStandardButton2
+End Sub
+Private Sub mStandardOptions_SelectionMade(ByVal ItemId As String, ByVal ItemIndex As Long)
+    SetButtonView ItemIndex, mStandardButton1, mStandardButton2
+End Sub
+Private Sub mShowAdvancedToggle_Toggled(ByVal IsPressed As Boolean)
+    this.CustomGroup.IsVisible = IsPressed
+End Sub
+
+Private Sub DefaultButtonAction(ByVal Button As RibbonButton)
+    MsgBox Button.Id & " Pressed", vbOKOnly Or vbInformation, MsgBoxTitle
+End Sub
+
+Private Sub SetButtonView(ByVal SelectedIndex As Integer, ParamArray Buttons())
+    Dim i As Long
+    Dim ShowLabel As Boolean: ShowLabel = ((SelectedIndex + 1) And 1) <> 0
+    Dim ShowImage As Boolean: ShowImage = ((SelectedIndex + 1) And 2) <> 0
```

Comment:
Please split on two separate lines, and add a comment to explain the magic hard-coded literals being bitwise-`And`ed here. If possible avoid bitwise logic in template VBA code beyond isolating flags in flag enum values.

Reply:
I hear you. This is just a quick-and-dirty demo of processing the DropDown index, which works more clearly in C# than in VBA. I will think about how better to code this in VBA, or evolve an alternative demonstration.